docs/building-apps/build-properties.md

```diff
@@ -916,6 +916,85 @@ only scan libraries with the `[LinkWith]` attribute for Objective-C classes:
 </PropertyGroup>
 ```
 
+## RunWithOpen
+
+This property determines whether apps are launched using the `open` command on
+macOS, or if the app's executable is executed directly.
+
+This only applies to macOS and Mac Catalyst apps.
+
+The default value is `true`. In this mode, the app will be launched by macOS as any other UI application, any stdout/stderr output will be swallowed by macOS, and the `dotnet run` command will finish as soon as the app has launched.
+
+If set to `false`, any stdout/stderr output will be printed to the current terminal, and the `dotnet run` command won't finish until the app has exited.
+
+The following properties can be used to configure the behavior when set to `true` (i.e. using the `open` command):
+
+### OpenNewInstance
+
+If a new instance will be opened if the app is already running (defaults to `false`).
+
+This will pass `-n` to `open` if set to `true`.
+
+Example:
+
+```shell
+$ dotnet run -p:OpenNewInstance=false
+```
+
+### RunEnvironment
```

Comment:
What about `UseEnviromentVars` or `UseEnviromentArgs`? `RunEnviroment` seems easy to clash just like the case we had with `NativeLibrary` just my 2 cents, naming is hard heh

Reply:
I removed this option completely, it's possible to do the same thing with the `OpenArguments` property, so document that instead.